buffering_manager.go

```diff
@@ -41,38 +51,63 @@ func initBufferingManager() {
 	internalBuffer = make(chan axRecords,
 		config.GetInt(analyticsBufferChannelSize))
 	closeBucketEvent = make(chan bucket)
+	doneInternalBufferChan = make(chan bool)
+	doneClosebucketChan = make(chan bool)
+
+	bucketMaplock.Lock()
 	bucketMap = make(map[int64]bucket)
+	bucketMaplock.Unlock()
 
 	// Keep polling the internal buffer for new messages
 	go func() {
 		for {
```

Comment:
same here. You can use `for records := range internalBuffer` and waive the explicit `more`